.github/workflows/cross-compile.yml

```diff
@@ -91,18 +99,18 @@ jobs:
 
       - name: Build for ${{ matrix.target }} in ${{ matrix.image }}
         run: |
-          export HOST_JAVA_HOME=$(dirname $(dirname $(readlink -f $(which java))))
-          export HOST_MAVEN_HOME=$(dirname $(dirname $(readlink -f $(which mvn))))
-          docker run --rm --workdir=/jssc \
-                --volume $PWD:/jssc \
-                --volume $HOST_JAVA_HOME:/jdk \
-                --volume $HOST_MAVEN_HOME:/mvn \
-                --env JAVA_HOME=/jdk \
+          docker run --rm --workdir=/work \
+                --volume $PWD:/work \
+                --volume $HOME/.m2:/root/.m2 \
                 --env MAVEN_OPTS=${MAVEN_OPTS} \
                 ${{ matrix.image }} \
                 bash -c \
-                  ' apt-get update && apt-get install --yes ${{ matrix.packages }} && \
-                  /mvn/bin/mvn -B clean install -P ${{ matrix.profile }} '
+                  'apt-get update && apt-get install --yes cmake maven openjdk-11-jdk-headless && \
+                   mvn -B clean install -P dockcross \
+                    -Dos.target.name=${{ matrix.os_target_name }} \
+                    -Dos.target.arch=${{ matrix.os_target_arch }} \
+                    -Dos.target.bitness=${{ matrix.os_target_bitness }} \
```

Comment:
`os.target.*` properties are normally set by the maven profile, and here they determine where the build lib file is put. I chose to create a single profile for Dockcross, and set these manually here - so that way old profiles still work as they used to, allowing devs to build the natives using unmodified toolchains.

Reply:
I like this idea.